src/app/content/highlights/hooks/printHighlights.ts

```diff
@@ -1,22 +1,25 @@
-import Sentry from '../../../../helpers/Sentry';
-import { addToast } from '../../../notifications/actions';
-import { toastMessageKeys } from '../../../notifications/components/ToastNotifications/constants';
 import { ActionHookBody, AppServices, MiddlewareAPI, Unpromisify } from '../../../types';
-import { actionHook, assertWindow } from '../../../utils';
+import { actionHook, assertWindow, CustomApplicationError } from '../../../utils';
 import { printSummaryHighlights, receiveSummaryHighlights, toggleSummaryHighlightsLoading } from '../actions';
+import { HighlightPopupPrintError } from '../errors';
 import { myHighlightsOpen } from '../selectors';
 import { loadMore, LoadMoreResponse } from './loadMore';
 
+let waitingForPromiseCollector = false;
```

Comment:
what is the purpose of this flag? global flag like this seems worrying to me

Reply:
`printHighlights` hook is calling `await services.promiseCollector.calm();` and for some reason that was causing infinite loop of waiting for itself if this hook returned the Promise
and since we want the error to be moved up we have to return the Promise so I've created this flag and replaced `await services.promiseCollector.calm();` with 
```
 if (waitingForPromiseCollector) {
    // wait for content to process/load
    await services.promiseCollector.calm();
    waitingForPromiseCollector = false;
  }
 ``` 
 I wanted to refactor this somehow, but I currently don't have more ideas on how to solve this

Reply:
all hook promises are added to the collector as a way to be able to tell when all logic has settled. so if you await collector.calm() in the hook it causes a dependency

from the previous version of this logic you could have remove the non-async hook wrapper, returned an async function as normal, you just have to remove the `await` before collector.calm() and move the subsequent logic into a `then` and not return that promise, so that it doesn't create a loop.

if there is any async logic after that you'll have to manually add it back to the collector, but in this case it looks like all synchronous logic so it doesn't matter